### ISSUE TITLE
Missing Column Data After Adding Left Join To Query in DuckDB Version 0.2.8

### ISSUE THREAD

**Issue by paul-aschenbrenner** [duckdb/duckdb#2102]
After upgrading to DuckDB version 0.2.8, column data is missing after adding a LEFT JOIN to a query.  This behavior is not seen in DuckDB version 0.2.7.

If you run the following query (queryOne) on the sample set in the attachment, there are 6219 rows which have a null value for the date field.  This is the expected value.  This is seen in both DuckDB versions 0.2.7 and 0.2.8.

```
SELECT one.id AS id,
    one.status AS schedule_status,
    one.date AS date
FROM 
    view_one AS one
```

If you run a second query (queryTwo) that includes a left join, it is expected that there will be the same 6219 rows which have a null value for the date field.  When the second query is run on the sample data set in the attachment, DuckDB version 0.2.8 has 6260 rows with a null date field instead of 6219 rows.  DuckDB version 0.2.7 will return the correct values for the date field and will have 6219 rows with a null date field.

```
SELECT one.id AS id,
    one.status AS schedule_status,
    one.date AS date
FROM 
    view_one AS one
LEFT JOIN
    view_two two ON two.id = one.id AND two.line = 1
```

I tested the sample Python script using Python version 3.9.5.

[DuckDB Missing Data in Query for 0.2.8.zip](https://github.com/duckdb/duckdb/files/6941094/DuckDB.Missing.Data.in.Query.for.0.2.8.zip)

**Comment by Mytherin** [comment]
Thanks for the report! I can confirm the issue. I will have a look.

**Comment by Mytherin** [comment]
Thanks again for the report! I have pushed a fix in #2110.
